### PR TITLE
Documentation for matrix_client/client.py

### PR DESCRIPTION
**Pull request by Half-Shot**

### CODE REVIEW

File: matrix_client/client.py [hunk 1]
```diff
@@ -52,6 +51,22 @@ def global_callback(incoming_event):
     """
 
     def __init__(self, base_url, token=None, valid_cert_check=True):
+        """ Create a new Matrix Client object.
+
+        Args:
+            base_url (string): The url of the HS preceding /_matrix.
```
Comment (erikjohnston) [inline matrix_client/client.py]: `str` is the official name

File: matrix_client/client.py [hunk 2]
```diff
@@ -52,6 +51,22 @@ def global_callback(incoming_event):
...
+                e.g. (ex: https://localhost:8008 )
+            token (Optional[string]): If you have an access token
```
Comment (erikjohnston) [inline matrix_client/client.py]: Is `Optional` actually a thing?

File: matrix_client/client.py [hunk 3]
```diff
@@ -84,23 +125,61 @@ def login_with_password(self, username, password, limit=1):
...
+            callback (func(roomchunk)): Callback called when an event arrives.
+        """
         self.listeners.append(callback)
 
     def listen_for_events(self, timeout=30000):
+        """ Listen once for events. Use listen_forever to block indefinitely.
+
+        Args:
+            timeout (int): How long to poll the Home Server for before
+                           retrying.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Be consistent with leading indentation. Choose either a single indent or to match them up (I generally find the former easier)

File: matrix_client/client.py [hunk 4]
```diff
@@ -115,10 +194,22 @@ def listen_for_events(self, timeout=30000):
                     listener(chunk)
 
     def listen_forever(self, timeout=30000):
+        """ Keep listening for events forever.
+
+        Args:
+            timeout (int): How long to poll the Home Server for before
+                           retrying.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Units? (Should the arg be called `timeout_ms`?)

File: matrix_client/client.py [hunk 5]
```diff
@@ -115,10 +194,22 @@ def listen_for_events(self, timeout=30000):
...
+        """
         while(True):
             self.listen_for_events(timeout)
 
     def start_listener_thread(self, timeout=30000):
+        """ Start a listener thread to listen for events in the background.
+
+        Args:
+            timeout (int): How long to poll the Home Server for before
+                           retrying.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Units.

File: matrix_client/client.py [hunk 6]
```diff
@@ -177,12 +278,25 @@ def _sync(self, limit=1):
             pass
 
     def get_user(self, user_id):
+        """ Return a User by their id.
+            NOTE: This function only returns a user object, it does not verify
+            the user with the Home Server.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Is the indentation here A Thing? Either way, there should be a blank line between summary and note

File: matrix_client/client.py [hunk 7]
```diff
@@ -177,12 +278,25 @@ def _sync(self, limit=1):
...
+        Args:
+            user_id (string): The matrix user id of a user.
+        """
         return User(self.api, user_id)
 
 
 class Room(object):
-
+    """ The Room class can be used to call room specific functions
+        after joining a room from the Client.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Shouldn't have indentation

File: matrix_client/client.py [hunk 8]
```diff
@@ -177,12 +278,25 @@ def _sync(self, limit=1):
...
+    """
     def __init__(self, client, room_id):
+        """ Create a blank Room object.
+            NOTE: This should ideally be called from within the Client.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Blank line needed

File: matrix_client/client.py [hunk 9]
```diff
@@ -296,21 +468,33 @@ def update_aliases(self):
...
         self.api = api
 
     def get_display_name(self):
-        '''See also get_friendly_name()
-        '''
+        """ Get this users display name.
+            See also get_friendly_name()
+
+        Returns:
+                string: Display Name
```
Comment (erikjohnston) [inline matrix_client/client.py]: Too much indentation

File: matrix_client/client.py [hunk 10]
```diff
@@ -296,21 +472,33 @@ def update_aliases(self):
 
 
 class User(object):
+"""
+The User class can be used to call user specific functions.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Spurious newline at the start

File: matrix_client/client.py [hunk 11]
```diff
@@ -114,20 +194,42 @@ def listen_for_events(self, timeout=30000):
                 for listener in self.rooms[chunk["room_id"]].listeners:
                     listener(chunk)
 
-    def listen_forever(self, timeout=30000):
+    def listen_forever(self, timeout_ms=30000):
+        """ Keep listening for events forever.
+
+        Args:
+            timeout_ms (int): How long to poll the Home Server for before
+                           retrying in milliseconds.
```
Comment (erikjohnston) [inline matrix_client/client.py]: Double indent.

**Comment by erikjohnston** [comment]
(Actual docs LGTM)

**Comment by erikjohnston** [comment]
You've also angered the gods of PEP8: http://matrix.org/jenkins/job/PythonSDK-PR/42/violations/file/matrix_client/client.py/

File: matrix_client/client.py (hunk 11 above)
Comment (SShrike) [inline matrix_client/client.py]: There seems to be an API change here?

**Pull request merged**